### PR TITLE
implementing new Utopia hooks

### PR DESCRIPTION
**Pull request by lohanidamodar**
## What does this PR do?

Implements Utopia Hook, that's newly introduced

## Test Plan

Existing tests should cover

## Related PRs and Issues

(If this PR is related to any other PR or resolves any issue or related to any issue link all related PR and issues here.)

### Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?

YES

### CODE REVIEW

File: app/executor.php [hunk 1]
```diff
@@ -581,57 +581,65 @@ function (string $runtimeId, array $vars, string $data, $timeout, $activeRuntime
...
-            break;
-        default:
-            $code = 500; // All other errors get the generic 500 server error status code
-    }
+App::error()
+    ->inject('utopia')
+    ->inject('error')
+    ->inject('request')
+    ->inject('response')
+    ->action(function ($utopia, $error, $request, $response) {
```
Comment (Meldiron) [inline app/executor.php:589]: Can we add types here?

File: composer.json [hunk 2]
```diff
@@ -42,7 +42,7 @@
         "ext-sockets": "*",
         "appwrite/php-clamav": "1.1.*",
         "appwrite/php-runtimes": "0.10.*",
-        "utopia-php/framework": "0.19.*",
+        "utopia-php/framework": "dev-master as 0.19.21",
```
Comment (Meldiron) [inline composer.json:45]: Let's not merge until the framework is released.

File: app/controllers/shared/api.php [hunk 3]
```diff
@@ -19,234 +19,267 @@
...
-    $isPrivilegedUser = Auth::isPrivilegedUser($roles);
-    $isAppUser = Auth::isAppUser($roles);
-
-    foreach ($timeLimitArray as $timeLimit) {
-        foreach ($request->getParams() as $key => $value) { // Set request params as potential abuse keys
-            if (!empty($value)) {
-                $timeLimit->setParam('{param-' . $key . '}', (\is_array($value)) ? \json_encode($value) : $value);
-            }
+App::init()
+    ->groups(['api'])
```
Comment (Meldiron) [inline app/controllers/shared/api.php:23]: The rest of PR puts groups under injection. Let's do it here too for consistency.

File: app/controllers/shared/api.php [hunk 4]
```diff
@@ -19,234 +19,267 @@
...
+            ->setParam('networkResponseSize', 0)
+            ->setParam('storage', 0)
+        ;
+
+        $deletes->setProject($project);
+        $database->setProject($project);
+    });
+
+App::init()
+    ->groups(['auth'])
```
Comment (Meldiron) [inline app/controllers/shared/api.php:135]: The rest of PR puts groups under injection. Let's do it here too for consistency.

File: app/controllers/shared/api.php [hunk 5]
```diff
@@ -19,234 +19,267 @@
...
+                break;
+
+            default:
+                throw new Exception('Unsupported authentication route', 501, Exception::USER_AUTH_METHOD_UNSUPPORTED);
+                break;
+        }
+    });
+
+App::shutdown()
+    ->groups(['api'])
```
Comment (Meldiron) [inline app/controllers/shared/api.php:189]: The rest of PR puts groups under injection. Let's do it here too for consistency.

**Review by Meldiron — APPROVED** [review]
LGTM, left tiny comments

File: app/executor.php (hunk 1 above)
Comment (lohanidamodar) [reply app/executor.php:589]: you mean for action? As the params are dynamic, I don't think that's easily possible?

Comment (eldadfux) [reply app/executor.php:589]: +1

File: app/controllers/shared/api.php (hunk 3 above)
Comment (eldadfux) [reply app/controllers/shared/api.php:23]: I think groups should be at the top.

**Review by eldadfux — COMMENTED** [review]
Looks good, once we fix comments made by @Meldiron we can merge.

**Review by eldadfux — APPROVED** [review]

**Pull request merged**